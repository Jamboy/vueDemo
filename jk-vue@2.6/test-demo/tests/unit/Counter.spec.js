/*
 * @Description: 
 * @Author: Jamboy
 * @Date: 2021-08-30 15:24:58
 * @LastEditTime: 2021-08-30 15:42:06
 */
import { mount } from "@vue/test-utils";
import Counter from "@/components/Counter.vue";

// 定义测试集
describe("Counter.vue", () => {
  let isCalled = false;
  const wrapper = mount(Counter,{
    listeners:{
      change(){
        isCalled = true;
      }
    }
  });
  it("render counter.html", () => {
    expect(wrapper.html()).toMatchSnapshot()
  });

  it("counter++",()=>{
    const button = wrapper.find('button')
    button.trigger('click')
    expect(wrapper.vm.count).toBe(1)
    expect(isCalled.toBe(true))
  });
});
