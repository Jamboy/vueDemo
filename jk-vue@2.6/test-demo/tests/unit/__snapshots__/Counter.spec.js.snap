// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue render counter.html 1`] = `<div><span>count:0</span> <button>count++</button></div>`;
